internal/packages/conditions.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package packages
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Masterminds/semver"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/multierror"
+)
+
+const kibanaVersionRequirement = "kibana.version"
+
+type packageRequirements struct {
+	kibana struct {
+		version *semver.Version
+	}
+}
+
+// CheckConditions method compares the given values with conditions in manifest.
+// The method is useful to check in advance if the package is compatible with particular stack version.
+func CheckConditions(manifest PackageManifest, keyValuePairs []string) error {
```

Comment:
I really like how generic this function is. I have been thinking of adding more conditions as part of https://github.com/elastic/package-registry/issues/670 and I think something like what you have here will come in very handy!